moc_openshift.py

```diff
@@ -340,6 +359,59 @@ def update_user_role_project(
             mimetype="application/json",
         )
 
+    def replace_moc_quota(self, project_name, new_quota):
+        """This will delete all resourcequota objects in a project and create new ones based on the new_quota specification"""
+        quota_def = self.get_quota_definitions()
+        if "QuotaMultiplier" in new_quota["Quota"]:
+            quota_multiplier = new_quota["Quota"]["QuotaMultiplier"]
+            for quota in quota_def:
+                quota_def[quota]["value"] = (
+                    quota_def[quota]["coefficient"] * quota_multiplier
+                    + quota_def[quota]["base"]
+                )
+                if "units" in quota_def[quota]:
+                    quota_def[quota]["value"] = (
+                        str(quota_def[quota]["value"]) + quota_def[quota]["units"]
+                    )
+        # RBB: flesh out
+        # else:
+        #
+        # RBB  need to overwrite the value in the quotadef with the ones from the new_quota
+
+        delete_resp = self.delete_moc_quota(project_name)
+        if delete_resp.status_code not in [200, 201]:
+            return Response(
+                response="Unable to delete current quotas in {project_name}\n {delete_resp.status}",
+                status=delete_resp.status_code,
+                mimetype="application/json",
+            )
+        create_resp = self.create_shift_quotas(project_name, quota_def)
+        if create_resp.status_code not in [200, 201]:
+            return Response(
+                response="Unable to define Quotas",
+                status=400,
+                mimetype="application/json",
+            )
+        return Response(
+            response="MOC Quotas Replaced",
+            status=200,
+            mimetype="application/json",
+        )
+
+    def get_quota_definitions(self):
+        with open("/app/quota/quota", "r") as file:
```

Comment:
In line with some other changes I'm making, I'd like to see this path come from an environment variable (e.g. `ACCT_MGR_QUOTA_FILE`). I'd like to make it easier to run this code locally (i.e. without deploying it anywhere), and removing the hardcoded paths is a necessary step towards that goal.

Reply:
I'll agree to the path, but I really don't like requiring bunches of environment variables.

I understand the desire to deploy it anywhere.

Perhaps we need a configuration file that contains things like this, the appropriate username/password to use for basic authentication ... .

Reply:
> I'll agree to the path, but I really don't like requiring bunches of environment variables.

I think you misunderstnd how one typically uses environment variables.

This makes it easy to configure via Secrets or ConfigMaps in kubernetes, and using things like `python-dotenv` (which is supported by Flask, for example), this lets you configure them via  `.env` file.

Reply:
If you're not already familiar with the concept of a 12 factor app, https://12factor.net/ is worth reading, which among other things has this to say:

> The twelve-factor app stores config in environment variables (often shortened to env vars or env). Env vars are easy to change between deploys without changing any code; unlike config files, there is little chance of them being checked into the code repo accidentally; and unlike custom config files, or other config mechanisms such as Java System Properties, they are a language- and OS-agnostic standard.

Reply:
For one, kubernetes has a standard place that it puts the service account token and namespace of the project - I don't quite understand why you want to replicate this in environmental variables other than some perceived convenience.  I can see specifying the location of such files in an environmental variable.

I am not altogether opposed using environmental variables, and yes, every application that I have worked has used them, some better than others.  My preference is that we use them sparingly and we pass in all of the information that the function requires as opposed to using global variables as implicit parameters.  But this is just a preference.

And the 12-factor app rules are reasonable, especially #10, keeping development, staging and production as similar as possible.  The worse application I have worked on were ones that had very difference environments for each, or were missing one or more of them (missing staging or doing development in production).

Reply:
I got that and will add the environment variable for the path.